docs/Using-Operations-Service.md

```diff
@@ -170,14 +170,26 @@ After that, you can produce an off-line signature using the following code:
 
 ```kotlin
 @Throws
-fun approveQROperation(operation: QROperation, password: String): String {
-    val authentication = PowerAuthAuthentication()
-    authentication.usePossession = true
-    authentication.usePassword = password
-    return operationsService.authorizeOfflineOperation(operation, authentication)
+fun approveQROperation(operation: QROperation, password: String?): String {
+    val auth = PowerAuthAuthentication()
+    auth.usePossession = true
+    if (password != null) {
+        // if a password is passed, use it as the second factor
+        auth.usePassword = password
+    } else if (operation.flags.biometryAllowed) {
```

Comment:
I don't like this example very much. Could we explain the logic in a series of simpler examples?

1. Check if the biometry is allowed for offline operation.
2. If yes, show the biometric approval.
3. If no, ask user for password and use a password approval.

Reply:
Improved in https://github.com/wultra/mtoken-sdk-android/pull/45/commits/4fd610e503b9ae1438f9982c1caccdc34293f03c